### ISSUE TITLE
Missing type map configuration or unsupported mapping

### ISSUE THREAD

**Issue by minhn** [AutoMapper/AutoMapper#869]
I'm experiencing some strange behavior with Automapper 4.0.4 Nuget package that I can't seems to pin point the issue.

Here's the detail of the exception.

Message:

>Missing type map configuration or unsupported mapping.

>Mapping types:
>ClassB -> Guid
>ConsoleApplication.ClassB -> System.Guid

>Destination path:
>ClassA

>Source value:
>ConsoleApplication.ClassB

Context:
(Not sure why Automapper is trying to map ClassB to Guid)

>{Trying to map ClassB to Guid.}

Context Parent:
>{Trying to map ClassB to ClassA.}

Code to reproduce the error:

```C#
public static void Main(string[] args)
{
    Mapper.CreateMap<ClassA, ClassB>()
        .ForMember(dest => dest.ClassB_Id, opt => opt.MapFrom(src => src.ClassA_Id))
        .ReverseMap()
        .ForMember(dest => dest.ClassA_Id, opt => opt.MapFrom(src => src.ClassB_Id));

    Mapper.AssertConfigurationIsValid(); //assertion passed
    
    var classA_1 = new ClassA {ClassA_Id = Guid.NewGuid()};
    var classB_1 = Mapper.Map<ClassB>(classA_1); //this works fine

    var classB_2 = new ClassB {ClassB_Id = Guid.NewGuid()};
    var classA_2 = Mapper.Map<ClassA>(classB_2); //this throws exception
}

public class ClassA
{
    public Guid ClassA_Id { get; set; }

    public ClassA()
    {
     }

    public ClassA(Guid id)
    {
        ClassA_Id = id;
    }
}

public class ClassB
{
    public Guid ClassB_Id { get; set; }
}

Work around that seems to work.
1) Name both Guid properties the same. i.e. ClassA.Id & ClassB.Id
2) Remove the parameterized constructor in ClassA
3) Downgrade to 3.3.1

**Comment by jbogard** [comment]
Have you tried the prerelease version on MyGet? This might be fixed there (the ctor issue).

**Comment by minhn** [comment]
@jbogard I'll give that a try thanks! any idea when the next stable release is?

**Comment by jbogard** [comment]
Just a few issues left to close

**Comment by minhn** [comment]
Thank you, this is fixed in 4.1.0 prerelease
